### PR TITLE
Add observability metrics for PowerScale

### PR DESCRIPTION
**Pull request by P-Cao**
# Description
Add observability metrics for PowerScale

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/488|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] UT
    PASS
```
~/csm-operator/controllers# go test ./... -coverprofile cover.out
ok      github.com/dell/csm-operator/controllers        0.730s  coverage: 88.6% of statements
~/csm-operator/pkg/modules# go test ./... -coverprofile cover.out
ok      github.com/dell/csm-operator/pkg/modules        0.130s  coverage: 89.3% of statements
```

- [x] e2e
  PASS
```
Tests:
--------------------------------------------------------------
1. TestCase: VolumeIoSuite
            Started:   2022-10-18 23:15:17.4351608 -0400 -0400
            Ended:     2022-10-18 23:17:42.531959242 -0400 -0400
            Result:    SUCCESS

            Stage metrics:
                    PVCAttachment:
                        Avg: 22.842835543s
                        Min: 6.544071162s
                        Max: 39.141599925s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCAttachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCAttachment_boxplot.png
                    PVCBind:
                        Avg: 9.680700677s
                        Min: 8.832446786s
                        Max: 10.528954568s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCBind.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCBind_boxplot.png
                    PVCCreation:
                        Avg: 1m3.68881545s
                        Min: 41.985988825s
                        Max: 1m25.391642075s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCCreation_boxplot.png
                    PVCDeletion:
                        Avg: 21.476539ms
                        Min: 16.909508ms
                        Max: 26.04357ms
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCDeletion_boxplot.png
                    PVCUnattachment:
                        Avg: 10.798936548s
                        Min: 6.36200046s
                        Max: 15.235872636s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCUnattachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PVCUnattachment_boxplot.png
                    PodCreation:
                        Avg: 35.157570585s
                        Min: 15.399487697s
                        Max: 1m20.485918142s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PodCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PodCreation_boxplot.png
                    PodDeletion:
                        Avg: 1.419191333s
                        Min: 942.561274ms
                        Max: 1.827422881s
                        Histogram:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PodDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/PodDeletion_boxplot.png
                        EntityNumberOverTime:
        /root/.cert-csi/reports/test-run-b05ad0c7/VolumeIoSuite9/EntityNumberOverTime.png

[2022-10-18 23:17:43]  INFO Avg time of a run:   130.66s
[2022-10-18 23:17:43]  INFO Avg time of a del:   12.03s
[2022-10-18 23:17:43]  INFO Avg time of all:     145.10s
[2022-10-18 23:17:43]  INFO During this run 100.0% of suites succeeded
STEP:      Executing  Enable forceRemoveDriver on CR
STEP:      Executing  Delete resources
STEP: Ending: Install PowerScale Driver(Standalone)

STEP: Starting: Install PowerScale Driver(With Authorization)
STEP: Required module not installed: authorization
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(With Authorization)

STEP: Starting: Install PowerScale Driver(Standalone), Enable Authorization
STEP: Required module not installed: authorization
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(Standalone), Enable Authorization

STEP: Starting: Install PowerScale Driver(With Replication)
STEP: Required module not installed: replication
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(With Replication)

STEP: Starting: Install PowerScale Driver(Standalone), Enable Replication
STEP: Required module not installed: replication
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(Standalone), Enable Replication

STEP: Starting: Uninstall PowerScale Driver
STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed
STEP:      Executing  Apply custom resources
Oct 18 23:17:48.161: INFO: Running '/usr/bin/kubectl --namespace=test-isilon apply --validate=true -f -'
Oct 18 23:17:48.461: INFO: stderr: ""
Oct 18 23:17:48.461: INFO: stdout: "containerstoragemodule.storage.dell.com/test-isilon created\n"
STEP:      Executing  Validate custom resources
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Enable forceRemoveDriver on CR
STEP:      Executing  Delete resources
STEP:      Executing  Validate [powerscale] driver is not installed
STEP: Ending: Uninstall PowerScale Driver

STEP: Starting: Install PowerScale Driver(With Authorization), Disable Authorization module"
STEP: Required module not installed: authorization
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(With Authorization), Disable Authorization module"

STEP: Starting: Install PowerScale Driver(With Replication), Disable Replication module"
STEP: Required module not installed: replication
STEP: Required module not installed, skipping
STEP: Ending: Install PowerScale Driver(With Replication), Disable Replication module"

STEP: Starting: Install PowerScale Driver(With Observability)
STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed
STEP:      Executing  Apply custom resources
Oct 18 23:18:13.542: INFO: Running '/usr/bin/kubectl --namespace=test-isilon apply --validate=true -f -'
Oct 18 23:18:13.771: INFO: stderr: ""
Oct 18 23:18:13.771: INFO: stdout: "containerstoragemodule.storage.dell.com/test-isilon created\n"
STEP:      Executing  Validate custom resources
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Validate [observability] module is installed
STEP:      Executing  Enable forceRemoveDriver on CR
STEP:      Executing  Delete resources
STEP: Ending: Install PowerScale Driver(With Observability)

STEP: Starting: Install PowerScale Driver(Standalone), Enable Observability
STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed
STEP:      Executing  Apply custom resources
Oct 18 23:18:28.853: INFO: Running '/usr/bin/kubectl --namespace=test-isilon apply --validate=true -f -'
Oct 18 23:18:29.076: INFO: stderr: ""
Oct 18 23:18:29.076: INFO: stdout: "containerstoragemodule.storage.dell.com/test-isilon created\n"
STEP:      Executing  Validate custom resources
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Validate [observability] module is not installed
STEP:      Executing  Enable [observability] module
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Validate [observability] module is installed
STEP:      Executing  Enable forceRemoveDriver on CR
STEP:      Executing  Delete resources
STEP: Ending: Install PowerScale Driver(Standalone), Enable Observability

STEP: Starting: Install PowerScale Driver(With Observability), Disable Observability module"
STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed
STEP:      Executing  Apply custom resources
Oct 18 23:18:44.203: INFO: Running '/usr/bin/kubectl --namespace=test-isilon apply --validate=true -f -'
Oct 18 23:18:44.420: INFO: stderr: ""
Oct 18 23:18:44.420: INFO: stdout: "containerstoragemodule.storage.dell.com/test-isilon created\n"
STEP:      Executing  Validate custom resources
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Validate [observability] module is installed
STEP:      Executing  Disable [observability] module
STEP:      Executing  Validate [powerscale] driver is installed
STEP:      Executing  Validate [observability] module is not installed
STEP:      Executing  Enable forceRemoveDriver on CR
STEP:      Executing  Delete resources
STEP: Ending: Install PowerScale Driver(With Observability), Disable Observability module"


• [SLOW TEST:242.485 seconds]
[run-e2e-test]E2E Testing
/root/csm-operator/tests/e2e/e2e_test.go:110
  Running all test Given Test Scenarios
  /root/csm-operator/tests/e2e/e2e_test.go:111
------------------------------

Ran 1 of 1 Specs in 242.708 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS



```

- [x] Manual
Capacity panel
image

IO panel
image

### CODE REVIEW

File: samples/storage_csm_powerscale_v240.yaml [hunk 1]
```diff
@@ -327,3 +327,84 @@ spec:
             # Default value: "INFO"
             - name: "TOPOLOGY_LOG_LEVEL"
               value: "INFO"           
+
+        - name: otel-collector
```
Comment (taohe1012) [inline samples/storage_csm_powerscale_v240.yaml:331]: otel-collector is a separate component ?

Comment (P-Cao) [reply samples/storage_csm_powerscale_v240.yaml:331]: > otel-collector is a separate component ?

Yes, it's an individual requirement according to our tickets.

**Comment by P-Cao** [comment]
@YianZong Just improved the logic of reconcile function. Please review this PR.

**Review by YianZong — APPROVED** [review]

**Review by taohe1012 — APPROVED** [review]

**Review by baoy1 — APPROVED** [review]

**Review by forrestxia — APPROVED** [review]

**Comment by forrestxia** [comment]
@rensyct please help review this PR. thanks!

**Comment by bharathsreekanth** [comment]
@abhi16394 @JacobGros @jooseppi-luna  - Please help review this PR.

**Review by bharathsreekanth — APPROVED** [review]

**Comment by bharathsreekanth** [comment]
@P-Cao - could you please proceed and merge this PR? thanks.

**Review by rensyct — APPROVED** [review]

**Pull request merged**